server/server.go

```diff
@@ -327,6 +327,21 @@ func NewServer(opts *Options) (*Server, error) {
 	return s, nil
 }
 
+// ClientURL returns the URL used to connect clients. Helpful in testing
+// when we designate a random client port (-1).
+func (s *Server) ClientURL() (string, error) {
```

Comment:
Good, but you did not use it yet. Since it will possibly return an error, it will have to be a two-steps process. Not sure if this is better than current:
```
	nc, err := nats.Connect(fmt.Sprintf("nats://%s:%d", opts.Host, opts.Port))
```

Reply:
I thought I used it in a test. Could make it not return an error and just send back empty string.

Reply:
I can remove or change. I think it would be nice for some of the older tests.

Reply:
We can keep it, but then returning error makes sense, but makes it more annoying to use since we would have to have a 2 steps as opposed to pass the result directly to nats.Connect().

Reply:
We could return URL which then could be nil?

Reply:
Actually, I don't think you should test on server running or not. You should simply return the url formatted with opts.Host/Port.

Reply:
As a string or URL. I think string IMO.

Reply:
I vote for string.

Reply:
Wil fix.